gui/src/components/import_modal.rs

```diff
@@ -0,0 +1,61 @@
+use std::{thread, time};
+
+use api::PostieApi;
+
+use crate::{Gui, ImportMode};
+
+pub fn import_modal(gui: &mut Gui, ctx: &egui::Context) {
+    if let Ok(mut import_window_open) = gui.import_window_open.try_write() {
+        if *import_window_open == true {
+            egui::Window::new("Import Modal")
+                .open(&mut *import_window_open)
+                .show(ctx, |ui| {
+                    ui.label("Please copy and paste the file path to import");
+                    ui.horizontal(|ui| {
+                        ui.text_edit_singleline(&mut gui.import_file_path);
+                        if ui.button("Import").clicked() {
+                            let path = gui.import_file_path.to_owned();
+                            if let Ok(import_mode) = gui.import_mode.try_read() {
+                                let import_result_clone = gui.import_result.clone();
+                                match *import_mode {
+                                    ImportMode::COLLECTION => {
+                                        let collections_for_worker = gui.collections.clone();
+                                        _ = tokio::spawn(async move {
+                                            let res =
+                                                PostieApi::import_collection(&path).await.unwrap();
+                                            let mut data = import_result_clone.lock().unwrap();
+                                            *data = Some(res);
+                                        });
+                                        _ = tokio::spawn(async move {
+                                            let sleep = time::Duration::from_millis(100);
+                                            thread::sleep(sleep);
```

Comment:
not the most elegant solution, but I was fighting with an error when trying to perform the refresh_collections call in the other tokio::spawn block.

the way things are currently written, both new worker threads are created one after the other and the refresh_collections was completing before the parsing and saving was complete so the ui never updated. with an artificial delay on the refresh, it gives enough time to let the file parse and be saved to the db.

will revisit this later.